### DISCUSSION TITLE
Remove a package without removing it from pyproject.toml

### DISCUSSION THREAD

**Discussion by steakhutzeee** [pdm-project/pdm#3041] [Q&A] [answered]
Hi, is it possible to uninstall a package without removing it from the `pyproject.toml` file? For example if I want to uninstall a dev dependency from the venv but keep the `pyproject.toml` as is.

Thank you.

**Comment by pawamoy** [comment]
`pdm run pip uninstall -y dep`?

**Reply by steakhutzeee** [reply, marked as answer]
Given that I could not have pip installed in the venv this solution can not be marked as definitive. Found the solution on the discord channel. It’s `pdm sync —only-keep -prod`.

**Reply by g-kartik** [reply]
Can you please explain how to use this command to uninstall a specific package?

**Reply by frostming** [reply]
There shouldn't be such need. If so, it's a signal that you should move this dependency to a different group then can use `pdm sync --clean-unselected` with proper group selection to unselect and uninstall that group as a whole.
